docs/source/topics/data_types.rst

```diff
@@ -492,8 +495,92 @@ Example of creating an ``PortableCode``:
+
+.. important::
+
+    If the ``engine_command`` contains variables (such as in the ``singularity exec --bind $PWD:$PWD {image_name}`` example), it is crucial that the ``Computer`` needs to have the ``use_double_quotes`` setting set to ``True``.
+    By default, a ``Computer`` will use single quotes to escape command line arguments and so the ``$PWD`` would not be expanded.
+    If this wasn't defined as such when the ``Computer`` was created, it can be changed through the API:
+
+    .. code-block::
+
+        computer = load_computer('some-computer')
+        computer.set_use_double_quotes(True)
+
+.. note::
+
+    The container image can be already download and stored on the target computer or can reference an image on a container index, such as Docker Hub.
+    In the latter case, the first time that the containerized code is run, the image will be automatically pulled from the index, which may take a while if the image is large.
+
+
+.. _topics:data_types:core:code:installed:containerized:setup:
+
+Setup
+^^^^^
+
+A ``ContainerizedCode`` can be created through the CLI as well as the API.
+The following examples show how to setup running ``bash`` in a base Docker container through Singularity to be run on the ``Computer`` named ``some-computer``:
+
+.. tab-set::
+
+    .. tab-item:: CLI
+
+        .. code-block:: console
+
+            verdi code create core.code.containerized \
+                --non-interactive \
+                --label containerized-code \
+                --default-calc-job-plugin core.arithmetic.add \
+                --computer some-computer \
+                --filepath-executable "/bin/sh" \
+                --image-name "docker://alpine:3" \
+                --engine-command "singularity exec --bind $PWD:$PWD {image_name}"
+
+    .. tab-item:: API
+
+        .. code-block:: python
+
+            from aiida.orm import ContainerizedCode, load_computer
+
+            code = ContainerizedCode(
+                computer=load_computer('some-computer')
+                filepath_executable='/bin/sh'
+                image_name='docker://alpine:3',
+                engine_command='singularity exec --bind $PWD:$PWD {image_name}'
+            ).store()
+
+
+.. _topics:data_types:core:code:installed:containerized:support:
+
+Supported container technologies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``ContainerizedCode`` is compatible with a variety of containerization technologies:
+
+* `Singularity <https://singularity-docs.readthedocs.io/en/latest/>`__
+* `Sarus <https://sarus.readthedocs.io/en/stable/>`__
+* `NVIDIA/enroot <https://github.com/NVIDIA/enroot/>`__
+* `Shifter <https://github.com/NERSC/shifter>`__
+
+Using `Docker <https://www.docker.com/>`__ directly is currently not supported, since Docker has to run with root permissions and it cannot be launched as a normal MPI program to propagate execution context to the container application.
```

Comment:
I get the MPI part, but I don't understand what the permissions for running docker has to do with AiiDA compatibility?

Either the user has the rights to run docker or they don't - I don't think this is relevant for the integration.

```suggestion
Using the `Docker <https://www.docker.com/>`__ container runtime is currently not supported, since Docker cannot be launched as a MPI program to propagate execution context to the container application.
```

Reply:
@unkcpz fine to remove the mention of the root permissions?

Reply:
> I get the MPI part, but I don't understand what the permissions for running docker has to do with AiiDA compatibility?

This is for the file created inside the container and file generated inside the container. If in the `Dockerfile` no `USER` is specified, the container with run with the `UID=0` the root user. The file it created can not be cleaned up by clean workdir for example.

Reply:
For the time being, I have rephrased it as

```

Using `Docker <https://www.docker.com/>`__ directly is currently not supported because:

* Docker needs to be run as the root user and the files created in the working directory inside the container will be owned by root, which prevents AiiDA from deleting those files after execution.
* Docker cannot be launched as a normal MPI program to propagate execution context to the container application.

Support may be added at a later time.
```

is that correct and intelligible?

Reply:
That is the default case I'd say but there is workaround and not so exact. Maybe better use the phrase from their official documentation is better?

["The Docker daemon always runs as the root user."](https://docs.docker.com/engine/install/linux-postinstall/#:~:text=The%20Docker%20daemon%20always%20runs,members%20of%20the%20docker%20group.)

> The Docker daemon always runs as the root user and the files created in the working directory inside the container will usually be owned by root if no `uid` is not specified in image, which prevents AiiDA from deleting those files after execution.